src/Microsoft.AspNet.OData.Shared/EdmODataAPIHandler.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Collections.Generic;
+
+namespace Microsoft.AspNet.OData
+{
+
+    /// <summary>
+    /// Handler Class to handle users methods for create, delete and update.
+    /// This is the handler for data modification where there is no CLR type.
+    /// </summary>
+    internal abstract class EdmODataAPIHandler
+    {
+        /// <summary>
+        /// TryCreate method to create a new object.
+        /// </summary>
+        /// <param name="changedObject">Changed object which can be appied on creted object, optional</param>
+        /// <param name="createdObject">The created object (Typeless)</param>
+        /// <param name="errorMessage">Any error message in case of an exception</param>
+        /// <returns>The status of the TryCreate Method, statuses are <see cref="ODataAPIResponseStatus"/></returns>
+        public abstract ODataAPIResponseStatus TryCreate(IEdmChangedObject changedObject, out IEdmStructuredObject createdObject, out string errorMessage);
```

Comment:
I would prefer the return type be a structure that has both `ODataAPIResponseStatus` and `errorMessage` as properties. I think it makes the pattern of having both return values clearer. Do you have any thoughts on this?